### ISSUE TITLE
Categories in `feather` output format get randomly sorted

### ISSUE THREAD

**Issue by alexwalkercebm** [opensafely-core/cohort-extractor#548]
After switching from using CSV format (e.g. [here](https://github.com/opensafely/long-covid/tree/735de64e2da036f6b04b02fb82b30ac33208c1bf), the categories in my previously neatly (alphabetically) ordered [output table](https://github.com/opensafely/long-covid/blob/735de64e2da036f6b04b02fb82b30ac33208c1bf/released_outputs/emis/counts_table.csv) got mixed up. Presumably the order is determined by the order that the categories appeared in the patient level table, and it's therefore non-deterministic.

It would be convenient if the categories preserved their lexical order.

**Comment by evansd** [comment]
> Presumably the order is determined by the order that the categories appeared in the patient level table

Yes, this is exactly right. I was a bit worried about what memory consumption would look like when adding binary output formats, as the one advantage of CSV was that we could stream directly to disk without having to hold the entire cohort in memory. And building categories on the fly as data came in was one way of keeping memory consumption down.

However, I think it's fairly straightforward to reorder the categories after the fact and it looks like Pandas can do it in-place so shouldn't have much more impact than a slight pause before writing the data.

For my own reference, the code is just:
```py
categorical = df[column_name].cat
categorical.reorder_categories(
    categorical.categories.sort_values(),
    inplace=True,
)
